doc/api/cli.md

```diff
@@ -73,6 +73,42 @@ If this flag is passed, the behavior can still be set to not abort through
 [`process.setUncaughtExceptionCaptureCallback()`][] (and through usage of the
 `domain` module that uses it).
 
+### `--build-snapshot`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+When `--build-snapshot` is set, the process will genearte a V8
+start snapshot of the heap and write it to disk before exiting.
+If `--snapshot-blob` is not specified, the generated blob
+will be written, by default, to `snapshot.blob` in the current working
+directory. Otherwise it will be written to the path specified by
+`--snapshot-blob`.
+
+At the moment, it's only supported to snapshot a process launched to
+run an entry point script, and the script can only load public Node.js
+APIs via `require()`. Support for `import` and loading user-land
```

Comment:
At the moment, it can only load a subset of Node.js APIs. Is the goal to support all modules before merging this PR?

Reply:
My plan is to "support whatever necessary to run a sufficiently sophisticated application for the initial iteration", in this case the TypeScript compiler and the marked renderer, so I think the modules currently supported in this PR would be enough for an initial, experimental iteration. Support for more modules can be done in future PRs (it would also require real world usage to know what module needs more fixing to get the states synchronized properly, so IMO attempting to provide support for all modules in one PR won't be very realistic anyway..)

Reply:
Ok, no problem for me but in that case, I think we should put the list of supported core modules in the documentation.

Reply:
I did another review of the modules listed in mksnapshot.js below - AFAICT, most of them should work across serialization/deserialization, with the exception of streams - if the stream is closed in time before the snapshot is created, it works alright (so using process.stdout/stderr etc. is OK since we clean them up properly before exit). But if the user leaves a stream open for some reason, and if there's no opened handle to throw an error on, it becomes buggy after deserialization -I think I'll remove the streams from the list for now, and add a note mentioning this caveat. This will probably take some effort to address in future iterations.